src/ripple/rpc/Yield.h

```diff
@@ -27,19 +28,26 @@
 
 namespace ripple {
 
+class JobQueue;
 class Section;
 
 namespace RPC {
 
-/** Yield is a generic placeholder for a function that yields control of
-    execution - perhaps to another coroutine.
+/** See the README.md in this directory for more information about how
+    the RPC yield mechanism works.
+ */
 
-    When code calls Yield, it might block for an indeterminate period of time.
+/** Callback: do something and eventually return. */
+using Callback = std::function <void ()>;
```

Comment:
Where is the original state from where the coroutine eventually yielded? Does the coroutine need to keep the state that it later needs in its object?